standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/entities/CompactionInfoBase.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.metastore.txn.entities;
+
+public class CompactionInfoBase implements Comparable<CompactionInfoBase> {
```

Comment:
why do we need this? can't we just use 1 CompactionInfo object?

Reply:
Currently CompactionInfo is like an union of all compaction related information. In numerous cases only a fraction of the fields are loaded to save resources (`findNextToCompact`, `findReadyToClean`), and in some other cases like `findPotentialCompactions` in Initiator, several fields are not even applicable for the scenario. This results in a situation that Compaction info is almost never fully initialized and you have to be aware of the context where it is used: You need to know which fields are present and which not. While technically this is a fully working approach, I would prefer introduce some class hierarchy. For example `CompactionCandidate` is the entity used by Initiator. Having this object makes it much more explicit which information is used/available in the initialization phase, and which others are not yet present/applicable. This is the idea behind the introduction of these two new base classes.

Reply:
could we do this in a separate PR and focus here just on CompactionTxnHandler?

Reply:
Removed all CompactionInfo class structure related changes